lib/compat/wordpress-6.0/blocks.php

```diff
@@ -95,13 +97,11 @@ function gutenberg_build_query_vars_from_query_block( $block, $page ) {
 		if ( ! empty( $block->context['query']['taxQuery'] ) ) {
 			$query['tax_query'] = array();
 			foreach ( $block->context['query']['taxQuery'] as $taxonomy => $terms ) {
-				if ( ! empty( $terms ) ) {
-					$term_ids = array_map( 'intval', $terms );
-					$term_ids = array_filter( $term_ids );
-
+				if ( is_taxonomy_viewable( $taxonomy ) && ! empty( $terms ) ) {
```

Comment:
[is_taxonomy_viewable](https://developer.wordpress.org/reference/functions/is_taxonomy_viewable/) checks if the taxonomy is `publicly_queryable`.